pkg/image/ostree_disk.go

```diff
@@ -50,12 +53,27 @@ type OSTreeDiskImage struct {
 func NewOSTreeDiskImage(commit ostree.SourceSpec) *OSTreeDiskImage {
 	return &OSTreeDiskImage{
 		Base:         NewBase("ostree-raw-image"),
-		CommitSource: commit,
+		CommitSource: &commit,
+	}
+}
+
+func NewOSTreeContainerDiskImage(container container.SourceSpec, ref string) *OSTreeDiskImage {
```

Comment:
to tamp down on confusion we could name this one `NewFromContainerOSTreeDiskImage()` and rename the other one to be `NewFromOSTreeCommitOSTreeDiskImage()`. The words "Container" and "Image" are both getting overloaded a lot and when I first read this I could have though we were creating a Container Image.

Reply:
That's a fair point.  The name of the pipeline in the manifest is `ostree-raw-image` for both and the type they return is `OSTreeDiskImage` so maybe that could be a prefix.
I think I'll go with
`NewOSTreeDiskImageFromCommit` and `NewOSTreeDiskImageFromContainer`.

Reply:
I like those even better.